python/lsst/afw/image/basicUtils.py

```diff
@@ -0,0 +1,106 @@
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the LSST License Statement and 
+# the GNU General Public License along with this program.  If not, 
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""Application Framework image-related classes including Image, Mask and MaskedImage
+"""
+import math
+
+import numpy
+
+import lsst.utils.tests
+import lsst.afw.geom as afwGeom
+from . import imageLib
+
+__all__ = ["makeImageFromArray", "makeMaskFromArray", "makeMaskedImageFromArrays",
+    "assertWcssNearlyEqualOverBBox"]
+
+suffixes = {str(numpy.uint16): "U", str(numpy.int32): "I", str(numpy.float32): "F", str(numpy.float64): "D"}
+
+def makeImageFromArray(array):
+    """Construct an Image from a NumPy array, inferring the Image type from the NumPy type.
+    Return None if input is None.
+    """
+    if array is None: return None
+    cls = globals()["imageLib.Image%s" % suffixes[str(array.dtype.type)]]
+    return cls(array)
+
+def makeMaskFromArray(array):
+    """Construct an Mask from a NumPy array, inferring the Mask type from the NumPy type.
+    """
+    if array is None: return None
+    cls = globals()["imageLib.Mask%s" % suffixes[str(array.dtype.type)]]
+    return cls(array)
+
+def makeMaskedImageFromArrays(image, mask=None, variance=None):
+    """Construct a MaskedImage from three NumPy arrays, inferring the MaskedImage types from the NumPy types.
+    """
+    cls = globals()["imageLib.MaskedImage%s" % suffixes[str(image.dtype.type)]]
+    return cls(makeImageFromArray(image), makeMaskFromArray(mask), makeImageFromArray(variance))
+
+@lsst.utils.tests.inTestCase
+def assertWcssNearlyEqualOverBBox(testCase, wcs0, wcs1, bbox, maxDiffSky=0.01*afwGeom.arcseconds, maxDiffPix=0.01,
+    nx=5, ny=5, msg="WCSs differ"):
+    """Compare pixelToSky and skyToPixel for two WCS over a rectangular grid of pixel positions
+
+    @param[in] testCase  unittest.TestCase instance the test is part of
+    @param[in] wcs0  WCS 0 (an lsst.afw.image.Wcs)
+    @param[in] wcs1  WCS 1 (an lsst.afw.image.Wcs)
+    @param[in] bbox  boundaries of pixel grid over which to compare the WCSs (an lsst.afw.geom.Box2I or Box2D)
+    @param[in] maxDiffSky  maximum separation between sky positions computed using Wcs.pixelToSky
+        (an lsst.afw.geom.Angle)
+    @param[in] maxDiffPix  maximum separation between pixel positions computed using Wcs.skyToPixel
+    @param[in] nx  number of points in x for the grid of pixel positions
+    @param[in] ny  number of points in y for the grid of pixel positions
+    @param[in] msg  exception message prefix; details of the error are appended after ": "
+
+    @throw AssertionError if the two WCSs do not match sufficiently closely
+    """
+    bboxd = afwGeom.Box2D(bbox)
+    xList = numpy.linspace(bboxd.getMinX(), bboxd.getMaxX(), nx)
+    yList = numpy.linspace(bboxd.getMinY(), bboxd.getMaxY(), ny)
+    measDiffSky = (afwGeom.Angle(0), -1, -1) # (sky diff, pix pos)
```

Comment:
Good catch. Thanks. I had originally been saving x,y and RA/Dec separately and didn't fix it everywhere. I switched to "?" for initial value, because the value is only ever printed anyway.

I also added a test for nx, ny > 0